test/simperium/channel_test.js

```diff
@@ -416,6 +416,37 @@ describe( 'Channel', function() {
 			channel.handleMessage( 'c:' + JSON.stringify( [{error: 405, id: 'thing', ccids: ['abc']}] ) );
 		} );
 
+		it( 'should stop sending duplicate changes after receiving a 409', done => {
+			const change = {o: 'M', id: 'thing', sv: 1, ccid: 'duplicate', v: diff( {}, {key: 'value'} )};
+
+			channel.localQueue.queue( change );
+
+			channel.once( 'send', () => {
+				// we should sent out our change the first time
+				channel.on( 'error', done );
+				channel.localQueue.once( 'queued', () => done( 'Should not queue duplicate changes' ) );
+				channel.once( 'acknowledge', () => done() );
```

Comment:
I have no idea which of these should be `channel` vs `channel.localQueue` and `on` vs. `once` and blah blah blah

Reply:
For testing purposes _ideally_ `localQueue` is hidden and we never know it exists. But given the existing codebase that might not be feasible without fixing a lot of the complexities.

Reply:
And `once` is the same as `on` except it removes itself as a listener after a single `.emit` that it is subscribed to.

The problem is that if a different test uses the same `channel` instance and subscribes to an event, it's possible that it could cause failures or unexpected handler executions.

Usually the end result being `done executed more than once` reported by the test runner.

Reply:
In some cases I need to listener to _fail_ the test should it be called, which means that even `once` leaves a lingering listener. with a new `Channel` in the `beforeEach` does that mean it won't matter in practice if each `it()` test leaves those listeners?